dbt_worker/tasks.py

```diff
@@ -1 +1,128 @@
+    # TODO: Add callback logic.
+
+
+def _invoke_runner(
+    task: Any, task_id: str, command: List[str], callback_url: Optional[str]
+):
+    """Invokes dbt runner with `command`, update task state if any exception is
+    raised.
+
+    Args:
+        task: Celery task.
+        task_id: Task id, it's required to update task state.
+        command: Dbt invocation command list.
+        callback_url: If set, if core raises any error, a callback will be
+            triggered."""
+    try:
+        dbt = dbtRunner()
+        _, _ = dbt.invoke(command)
+    except Exception as e:
+        _update_state(
+            task,
+            task_id,
+            FAILURE,
+            {"exc_type": type(e).__name__, "exc_message": str(e)},
+            callback_url,
+        )
+
+
+def _get_task_status(task: Any, task_id: str):
+    """Retrieves task state for `task_id` from task backend."""
+    return task.AsyncResult(task_id).state
+
+
+def _insert_log_path(command: List[str], task_id: str):
+    """If command doesn't specify log path, insert default log path at start."""
+    # We respect user input log_path.
+    if is_command_has_log_path(command):
+        return
+    command.insert(0, LOG_PATH_ARGS)
+    command.insert(1, get_task_artifacts_path(task_id, None))
+
+
+def _invoke(task: Any, command: List[str], callback_url: Optional[str] = None):
+    """Invokes dbt command.
+    Args:
+        command: Dbt commands that will be executed, e.g. ["run",
+            "--project-dir", "/a/b/jaffle_shop"].
+        callback_url: String, if set any time the task status is updated, worker
+            will make a callback. Notice it's not complete, in some cases task
+            status may be updated but we are not able to trigger callback, e.g.
+            worker process is killed."""
+    task_id = task.request.id
+    _insert_log_path(command, task_id)
+    logger.info(f"Running dbt task ({task_id}) with {command}")
+    # TODO: Send callback to infer task start.
+
+    # To support abort, we need to run dbt in a child thread, make parent thread
+    # monitor abort signal and join with child thread.
+    t = Thread(target=_invoke_runner, args=[task, task_id, command, callback_url])
+    t.start()
+    while t.is_alive():
+        # TODO: Handle abort signal.
+        t.join(JOIN_INTERVAL_SECONDS)
+    # By the end of execution, a task state might be.
+    # - STARTED, everything is fine!
+    # - FAILURE, error occurs.
+    # - ABORTED, user abort the task.
+    task_status = _get_task_status(task, task_id)
+    if task_status == ABORTED:
+        # TODO: Handle abort.
+        pass
+    # If task status is not propagatable, we need to mark it as success manually
+    # to trigger callback.
+    elif task_status not in PROPAGATE_STATES:
+        _update_state(task, task_id, SUCCESS, {}, callback_url)
+    # Raises Ignore exception to make Celery not automatically set state to
+    # SUCCESS.
+    raise Ignore()
```

Comment:
Just a note for future reference-- spoke about this in person, we are going to verify whether celery has a config option to override default behavior wherein celery sets all completed tasks to success, to avoid using Ignore(). This is fine for now.